collect.sh

```diff
@@ -98,8 +99,15 @@ do
   fi
 
   #echo $TIME" 0 0 0 0"  >> "data/ref.log"
-  
-  sleep 1
+
+  END_TIME=$(date +%s.%N)
+  ELAPSED=$(echo "$END_TIME - $START_TIME" | bc)
```

Comment:
you don't have to use BC repository here. you can simply do this in bash do the arithmetic calculation
addition
```
$(($SUM_ALO_PID+$ALO_PID))
```
substation
```
$(($SUM_ALO_PID-$ALO_PID))
```

in your case, this should looks sth like this for example
```
ELAPSED=$(($END_TIME-$START_TIME))
```

Reply:
I set myself up for failure using `date +%s.%N`. How does it look now?